src/main.rs

```diff
@@ -75,7 +84,13 @@ async fn main() {
             }
         }
         None => {
-            await_ready(authority.clone(), backoff).await;
+            let ready = await_ready(authority.clone(), backoff);
+            if tokio::time::timeout(timeout, ready).await.is_err() {
+                eprintln!(
+                    "linkerd-proxy failed to become ready within {:?} timeout",
+                    timeout
+                );
+            }
```

Comment:
I think we should actively fail the command in this case. This will report an obvious PostStartError status in `kubectl`.

```suggestion
            let timeout = async move {
                if let Some(t) = timeout {
                    tokio::time::sleep(t).await;
                    t
                } else {
                    futures::future::pending().await
                }
            };
            tokio::select! {
                _ = ready => {},
                timeout = timeout => {
                    eprintln!(
                        "linkerd-proxy failed to become ready within {:?} timeout",
                        timeout
                    );
                    std::process::exit(EX_UNAVAILABLE);
                }
            }
```

We'll want to add an unavailable code (69) in here https://github.com/linkerd/linkerd-await/blob/806da1d3780111abfab9b590d1a2c140d0ec886d/src/main.rs#L52-L53

Reply:
I still used the `tokio::time::timeout` function when we have a `timeout`, but I added the exit code.